### PR TITLE
check_icmp: buffer offerflow

### PR DESCRIPTION
**Pull request by RincewindsHat**

### CODE REVIEW

File: plugins-root/check_icmp.c [hunk 1]
```diff
@@ -50,19 +50,24 @@ const char *email = "devel@monitoring-plugins.org";
 #if HAVE_SYS_SOCKIO_H
 #include <sys/sockio.h>
 #endif
+
+/*
```
Comment (sni) [inline plugins-root/check_icmp.c:54]: That's bad practice. Either remove it or keep it, but commenting out code should be avoided. That's what we have git for.

File: plugins-root/t/check_icmp.t [hunk 2]
```diff
@@ -35,51 +35,72 @@ my $hostname_invalid   = getTestParameter( "NP_HOSTNAME_INVALID",
 
 my $res;
 
+# 1
```
Comment (sni) [inline plugins-root/t/check_icmp.t:38]: what's the idea of these comments?

File: plugins-root/check_icmp.c [hunk 3]
```diff
@@ -1514,18 +1534,19 @@ get_threshold(char *str, threshold *th)
...
-icmp_checksum(unsigned short *p, int n)
+icmp_checksum(uint16_t *p, size_t n)
 {
 	unsigned short cksum;
 	long sum = 0;
 
-	while(n > 2) {
-		sum += *p++;
-		n -= sizeof(unsigned short);
+	/* sizeof(uint16_t) == 2 */
```
Comment (sni) [inline plugins-root/check_icmp.c:1542]: what's the reason of using hard coded 2 instead of calling sizeof?

Comment (RincewindsHat) [reply plugins-root/check_icmp.c:1542]: No specific reason, it's the same anyway and at list it is consistent with line 1545 now. Would it be better understandable, if I used `sizeof`?

File: plugins-root/t/check_icmp.t (hunk 2 above)
Comment (RincewindsHat) [reply plugins-root/t/check_icmp.t:38]: To bee honest, just to help me with counting them

Comment (sni) [reply plugins-root/t/check_icmp.t:38]: I would just run the test once after adding test cases. It then tells you the new number of tests and it can be changed.

File: plugins-root/check_icmp.c (hunk 3 above)
Comment (sni) [reply plugins-root/check_icmp.c:1529]: I'd rather change the other occurance as well. But that's only my opinion.

File: plugins-root/t/check_icmp.t (hunk 2 above)
Comment (RincewindsHat) [reply plugins-root/t/check_icmp.t]: Removed the comments

**Comment by RincewindsHat** [comment]
should be fixed, review?

File: plugins-root/check_icmp.c [hunk 4]
```diff
@@ -1514,18 +1521,19 @@ get_threshold(char *str, threshold *th)
...
```
Comment (sni) [inline plugins-root/check_icmp.c:1529]: can this comment be removed?

File: plugins-root/check_icmp.c [hunk 5]
```diff
@@ -1514,18 +1521,19 @@ get_threshold(char *str, threshold *th)
...
+	while(n >= 2) {
+		sum += *(p++);
+		n -= 2;
 	}
 
 	/* mop up the occasional odd byte */
-	if(n == 1) sum += (unsigned char)*p;
+	if(n == 1) sum += *((uint8_t *)p -1);
```
Comment (sni) [inline plugins-root/check_icmp.c:1536]: should this be uint16_t?

File: plugins-root/check_icmp.c [hunk 6]
```diff
@@ -913,9 +902,27 @@ wait_for_reply(int sock, u_int t)
...
+					printf("%0.3f ms rtt from %s, outgoing ttl: %u, incoming ttl: %u, max: %0.3f, min: %0.3f\n",
+						(float)tdiff / 1000,
+						address,
+						ttl,
+						ip->ip.ip_ttl,
+						(float)host->rtmax / 1000,
+						(float)host->rtmin / 1000);
+					break;
+					  };
+				case AF_INET6: {
```
Comment (sni) [inline plugins-root/check_icmp.c:917]: what's the reason for this change?

**Review by sni — COMMENTED** [review]
i've got some questions :-)

File: plugins-root/check_icmp.c (hunk 5 above)
Comment (RincewindsHat) [reply plugins-root/check_icmp.c:1536]: I don't think so, this should catch a single byte, but I might be thinking wrong.

File: plugins-root/check_icmp.c (hunk 6 above)
Comment (RincewindsHat) [reply plugins-root/check_icmp.c:917]: I think, the problem is,, that we don't get an incoming ttl if IPv6 is used, due to differences in the data structure (or the protocol?). But I already forgot was the specifics were there, I can research them again, if you would like to know the details.

**Review by sni — APPROVED** [review]
